packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -2993,64 +2999,74 @@ function flushRenderPhaseStrictModeWarningsInDEV() {
   }
 }
 
-function commitDoubleInvokeEffectsInDEV(
-  fiber: Fiber,
+function recursivelyTraverseAndDoubleInvokeEffectsInDEV(
+  root: FiberRoot,
+  parentFiber: Fiber,
+  isInStrictMode: boolean,
   hasPassiveEffects: boolean,
 ) {
-  if (__DEV__ && enableStrictEffects) {
-    // TODO (StrictEffects) Should we set a marker on the root if it contains strict effects
-    // so we don't traverse unnecessarily? similar to subtreeFlags but just at the root level.
-    // Maybe not a big deal since this is DEV only behavior.
+  let child = parentFiber.child;
```

Comment:
We should wrap this whole function in `if (parentFiber.subtreeFlags & PlacementDEV)`. That way we can skip over any paths that don't include insertions.